### PR TITLE
(#138) - Support for canceling upload

### PR DESCRIPTION
**Pull request by andreoss**
Per #138 

-

### CODE REVIEW

File: src/main/java/com/artipie/docker/http/UploadEntity.java [hunk 1]
```diff
@@ -456,4 +461,59 @@ private BlobCreatedResponse(final RepoName name, final Digest digest) {
...
+        public Response response(
+            final String line,
+            final Iterable<Map.Entry<String, String>> headers,
+            final Publisher<ByteBuffer> body
+        ) {
+            final Request request = new Request(line);
+            final RepoName name = request.name();
+            final String uuid = request.uuid();
+            return new AsyncResponse(
+                this.docker.repo(name).uploads().get(uuid).thenApply(
```
Comment (g4s8) [inline src/main/java/com/artipie/docker/http/UploadEntity.java:501]: You can use `.thenCompose` to join `CompletableFuture`

File: src/main/java/com/artipie/docker/http/UploadEntity.java [hunk 2]
```diff
@@ -456,4 +461,59 @@ private BlobCreatedResponse(final RepoName name, final Digest digest) {
...
+                    found -> found.<Response>map(
+                        upload -> new AsyncResponse(
```
Comment (g4s8) [inline src/main/java/com/artipie/docker/http/UploadEntity.java:503]: No need to wrap with second `AsyncReponse`

File: src/main/java/com/artipie/docker/http/UploadEntity.java [hunk 3]
```diff
@@ -456,4 +461,59 @@ private BlobCreatedResponse(final RepoName name, final Digest digest) {
...
+                            upload.cancel().thenApply(
+                                offset -> new RsWithHeaders(
+                                    new RsWithStatus(RsStatus.OK),
+                                    new ContentLength("0"),
```
Comment (g4s8) [inline src/main/java/com/artipie/docker/http/UploadEntity.java:507]: Is it required by Docker API spec to put zero content length here?

**Review by g4s8 — CHANGES_REQUESTED** [review]
@andreoss thanks, please check my comments

File: src/main/java/com/artipie/docker/http/UploadEntity.java (hunk 3 above)
Comment (andreoss) [reply src/main/java/com/artipie/docker/http/UploadEntity.java]: @g4s8 No. Removed.

**Review by g4s8 — APPROVED** [review]
@andreoss thanks

**Pull request merged**